### PR TITLE
Create elapsed_time_calculator.js

### PR DESCRIPTION
**Pull request by jeelanp2003**
Add an example for elapsed_time_calculator PL/SQL equivalent

### CODE REVIEW

**Comment by jeelanp2003** [comment]
Add an example for elapsed_time_calculator PL/SQL equivalent

File: elapsed_time_calculator.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    for (i=1; i<=50000; i++){
+        var selectquery = SELECT 1 INTO n FROM dual; // This is work-in-progress.
+        for (const row of selectquery) {
+        }
+    }
+
+    var end = new Date().getTime();
+    COMMIT WORK;
+
+    return end - start;
```
Comment (abhijpes) [inline elapsed_time_calculator.js]: would be good to enclose within brackets: `(end - start)` .

File: elapsed_time_calculator.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// NOTE: THIS IS STILL A WORK-IN-PROGRESS AND NOT YET TESTED FULLY 
+
+function calculate_elapsed_time() {
+    var n = 0;
+
+    BEGIN WORK;
+    var start = new Date().getTime();
+
+    for (i=1; i<=50000; i++){
+        var selectquery = SELECT 1 INTO n FROM dual; // This is work-in-progress.
```
Comment (abhijpes) [inline elapsed_time_calculator.js]: This query doesn't make sense as there is no `dual`. Simplest valid N1QL select query is `SELECT 1` which provides a single row result.

File: elapsed_time_calculator.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        for (const row of selectquery) {
+        }
```
Comment (abhijpes) [inline elapsed_time_calculator.js:72]: Would be best to add comments within the loop on some random execution:

```
        for (const row of selectquery) {
             // application logic
        }
```

File: elapsed_time_calculator.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+/
+*/
+
+// Below JSUDF calculates elapsed time down to the 1000th of a second (as opposed to 100th in PL/SQL implementation above)
+// NOTE: THIS IS STILL A WORK-IN-PROGRESS AND NOT YET TESTED FULLY 
+
+function calculate_elapsed_time() {
+    var n = 0;
+
+    BEGIN WORK;
```
Comment (abhijpes) [inline elapsed_time_calculator.js:65]: Not sure whether query txns are required for this use case. Txns will be useful if the UDF is running DML queries. In this example there are only select statements.

Comment (jeelanp2003) [reply elapsed_time_calculator.js:65]: Yeah, I had the same doubts. Will check with Kamini

File: elapsed_time_calculator.js (hunk 1 above)
Comment (jeelanp2003) [reply elapsed_time_calculator.js]: Done

File: elapsed_time_calculator.js (hunk 3 above)
Comment (jeelanp2003) [reply elapsed_time_calculator.js:72]: done

File: elapsed_time_calculator.js (hunk 2 above)
Comment (jeelanp2003) [reply elapsed_time_calculator.js]: done

**Review by abhijpes — APPROVED** [review]
+1 with the changes and maybe with txn statements removed.

**Pull request merged**